migtools/oadp-non-admin
# fix: Update NAB status phase and condition together

## Why the changes were made

As discussed with the team, we want to do less reconcile steps for NAB full reconcile loop (but keeping only making one update call per reconcile). With this PR, NAB reconcile update phases and conditions together in one update call.

Also, simplified Predicate and Handlers logic.

## How to test the changes made

Run tests and check logs (`make simulation-test`) or test in a cluster following [development documentation](https://github.com/migtools/oadp-non-admin/blob/master/docs/CONTRIBUTING.md#install-from-source).
